doc-2285 feat(configuration-parameters.backup): support Azure blob (3.11, 4.1.1, 4.2) (#541)
tigergraph/server-docs: modules/backup-and-restore/pages/configurations.adoc

|===
|Configuration parameter |Description |Default

|System.Backup.Local.Enable |Whether to store the database backup data to a local path.
If this parameter is set to true, `System.Backup.S3.Enable` cannot be set to true.
|`false`

|System.Backup.Local.Path |Local path to store the backup files.
Required if backup is to be stored locally. | `null`

> |System.Backup.S3.AWSAccessKeyID |The AWS access key ID used for the S3 backup bucket. It corresponds to the "Access Key" in GCS and the "Account Name" in Azure Blob, respectively. | `null`

Reviewer: ```suggestion
|System.Backup.S3.AWSAccessKeyID |The access key for the cloud storage bucket. For AWS, the S3 access key ID; for GCS, the "Access Key". For Azure, the blob "Account Name". | `null`
```

> |System.Backup.S3.AWSSecretAccessKey |The secret access key used for the AWS S3 bucket. It corresponds to the "Secret" in GCS and the "Account Key" in Azure Blob, respectively.

Reviewer: ```suggestion
|System.Backup.S3.AWSSecretAccessKey |The {secret access key, secret, secret} for the {AWS S3, Azure Blob, GCS} bucket, respectively.
```

*NOTE*: If setting this in interactive mode, store the key in a file and provide the path to the file, e.g., `@/tmp/test_secret`.
|`+null+`

> |System.Backup.S3.BucketName |The S3 bucket name. It corresponds to the "Bucket" in GCS and the "Container" in Azure Blob, respectively.|`null`

Reviewer: ```suggestion
|System.Backup.S3.BucketName |The {bucket, container, bucket} name for {AWS, Azure, GCS}, respectively.|`null`
```

The specific storage service used for the backup is determined by the `System.Backup.S3.Endpoint` configuration setting. 

If this parameter is set to true, `System.Backup.Local.Enable` cannot be set to true.|`+false+`

|System.Backup.S3.Endpoint 
a| A fully qualified URI, such as https://s3.amazonaws.com/, 
is used to specify the storage service’s endpoint for the client.
It is used in the following situations:

* To backup to a private AWS S3, typically set it to "https://s3.amazonaws.com/" or any available URI.
* To backup to an S3-compatible service, set it to its corresponding service URI.
* To backup to Google Cloud Storage (GCS), set it to "https://storage.googleapis.com"
* To backup to Azure Blob, set it to "https://<your storage account name>.blob.core.windows.net"." +

Except for the above specific situations, leave it empty.| `null`

|System.Backup.TimeoutSec |Timeout limit for the backup operation in seconds |`+18000+`

|System.Backup.CompressProcessNumber | Number of concurrent processes for compression during backup. `0` means the number of processes used to compress is equal to the number of CPU cores on each node.

> It's recommended to keep the default value `10`.

Reviewer: ```suggestion
We recommending keeping the default value `10`.
```

| "DefaultCompression"
|===

> IMPORTANT: If `System.Backup.Local.Enable` is set to `true`, this also enables a daily full backup at 12:00am UTC.

Author: the daily backup is only supported by  our own cloud platform. it is not available for normal customers. remove this line to avoid misleading

After configuring the parameters, run `gadmin config apply` to apply the new parameter values.

== Configure System.Backup.S3.Endpoint

Typically, there's no need to configure the `System.Backup.S3.Endpoint` parameter on a TigerGraph Server.
This is because the system auto-detects the regional endpoint for AWS S3 backups. 

.Users should configure this parameter *only* for special cases, such as:
* To backup to a private AWS S3, typically set it to "https://s3.amazonaws.com/" or any available URI.
* To backup to an S3-compatible service, set it to its corresponding service URI.
* To backup to Google Cloud Storage (GCS), set it to "https://storage.googleapis.com"
* To backup to Azure Blob, set it to "https://<your storage account name>.blob.core.windows.net"."

Except for the above specific situations, leave it empty.

== Backup to AWS S3

> To configure backup files to an AWS S3 Bucket for an on-premise TigerGraph Server cluster, users need to complete the following steps:

Reviewer: ```suggestion
To configure backup files to an AWS S3 Bucket for an on-premises TigerGraph Server cluster, complete the following steps:
```

----
gadmin config apply -y
----

== Backup to Azure Blob

> Similar to AWS S3, once the Azure Blob Container is created and configured properly, for more information,see https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction[Introduction to Azure Blob Storage],

Reviewer: ```suggestion
Similar to AWS S3, once the Azure Blob Container is created and configured properly, then configure it to be your backup storage via the following steps.
You may find Microsoft's https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction[Introduction to Azure Blob Storage] to be helpful.
```

users only need to complete the configurations by the following steps:

. Enable storing backup data in S3
+
[console,]
----
gadmin config set "System.Backup.S3.Enable" "true"
----
. Specify the backup S3 Endpoint
+
[console,]
----
gadmin config set "System.Backup.S3.Endpoint" "https://<account-name>.blob.core.windows.net"
----
. Specify bucket name
+
[console,]
----
gadmin config set "System.Backup.S3.BucketName" "<container-name>"
----
. Set S3 backup AccessKeyID
+
[console,]
----
gadmin config set "System.Backup.S3.AWSAccessKeyID" "<account-name>"
----

. Set S3 backup SecretAccessKey
+
[console,]
----
gadmin config set "System.Backup.S3.AWSSecretAccessKey" "<account-key>"
----
. Apply the new parameter values
+

> .Restart all services

Author: It is unnecessary to restart all services if we only change the backup configurations. Running `gadmin config apply` is enough to make it effective.
